python/lsst/pipe/base/graphBuilder.py

```diff
@@ -1017,6 +1063,20 @@ def resolveDatasetRefs(
                     quantum.prerequisites[datasetType].update(
                         {ref.dataId: ref for ref in prereq_refs if ref is not None}
                     )
+
+                # Resolve all task inputs and outputs.
+                if idMaker:
+                    for datasetType in task.inputs:
+                        resolved = idMaker.resolveDict(quantum.inputs[datasetType])
+                        quantum.inputs[datasetType].update(resolved)
+                    for datasetType in task.outputs:
+                        resolved = idMaker.resolveDict(quantum.outputs[datasetType])
+                        quantum.outputs[datasetType].update(resolved)
+                    for inits in (task.initInputs, task.initOutputs):
```

Comment:
I haven't spotted why these two can be handled together but the inputs and outputs can't be.

Reply:
I thought that was the correct logic when I was writing it, but now I do not remember why 🙂 I'll try to unify that and see if that works.

Reply:
I replaced that with more unified code.